src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,31 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
```

Comment:
When a `DataProcessingException` is caught, it should be thrown with a descriptive message that includes the country details. This will help to identify the operation that failed and the reason for the failure. For example:
throw new DataProcessingException("Cannot add country " + country + " to DB", e);